athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/connection/EnvironmentProperties.java

```diff
@@ -0,0 +1,135 @@
+
+        HashMap<String, String> connectionEnvironment = new HashMap<>();
+        if (StringUtils.isNotBlank(glueConnectionName)) {
+            connectionEnvironment.put(DEFAULT_GLUE_CONNECTION, glueConnectionName);
+            Connection connection = getGlueConnection(glueConnectionName);
+            Map<String, String> connectionPropertiesWithSecret = new HashMap<>(connection.connectionPropertiesAsStrings());
+            connectionPropertiesWithSecret.putAll(authenticationConfigurationToMap(connection.authenticationConfiguration()));
+
+            connectionEnvironment.putAll(connectionPropertiesToEnvironment(connectionPropertiesWithSecret));
+            connectionEnvironment.putAll(athenaPropertiesToEnvironment(connection.athenaProperties()));
+        }
+
+        connectionEnvironment.putAll(lambdaEnvironment); // Overwrite connection environment variables with lambda environment variables
+        return connectionEnvironment;
+    }
+
+    public Connection getGlueConnection(String glueConnectionName) throws RuntimeException
+    {
+        try {
+            HashMap<String, String> lambdaEnvironment = new HashMap<>(System.getenv());
+            GlueClient awsGlue = GlueClient.builder()
+                    .httpClientBuilder(ApacheHttpClient
+                            .builder()
+                            .connectionTimeout(Duration.ofMillis(CONNECT_TIMEOUT)))
+                    .build();
+            if (lambdaEnvironment.getOrDefault("USE_GAMMA_GLUE", "false").equals("true")) {
+                awsGlue = GlueClient.builder()
+                        .endpointOverride(new URI(String.format("https://glue-gamma.%s.amazonaws.com", lambdaEnvironment.get("AWS_REGION"))))
+                        .httpClientBuilder(ApacheHttpClient
+                                .builder()
+                                .connectionTimeout(Duration.ofMillis(CONNECT_TIMEOUT)))
+                        .build();
+            }
+            GetConnectionResponse glueConnection = awsGlue.getConnection(GetConnectionRequest.builder().name(glueConnectionName).build());
+            logger.debug("Successfully retrieved connection {}", glueConnectionName);
+            return glueConnection.connection();
+        }
+        catch (Exception err) {
+            logger.error("Failed to retrieve connection: {}, and parse the connection properties!", glueConnectionName);
+            throw new RuntimeException(err.toString());
+        }
+    }
+
+    private Map<String, String> authenticationConfigurationToMap(AuthenticationConfiguration auth)
+    {
+        Map<String, String> authMap = new HashMap<>();
+
+        if (auth != null && StringUtils.isNotBlank(auth.secretArn())) {
+            String[] splitArn = auth.secretArn().split(":");
+            String[] secretNameWithRandom = splitArn[splitArn.length - 1].split("-"); // 6 random characters at end. at least length of 2
```

Comment:
why are we getting the secret name from the split? shouldn't this be already available on the fly? (aka from glue connection or lambda envs?)

Reply:
glue connection keeps track of secret arn which is in this format (https://docs.aws.amazon.com/secretsmanager/latest/userguide/whats-in-a-secret.html). Our connector secret logic has always used just the secret name part without the last 6 random characters. To keep our connector behavior consistent with old secret_name logic, I parse SecretArn until I get only the secret name.